### PR TITLE
Support int64 index dtype in sparse logistic regression

### PR DESCRIPTION
**Pull request by lijinf2**
Meanwhile update the test environment to cuml 24.08 nightly.

### CODE REVIEW

**Comment by lijinf2** [comment]
build

File: python/tests/conftest.py [hunk 1]
```diff
@@ -63,7 +63,7 @@ def tmp_path() -> Generator[str, None, None]:
     "spark.python.worker.reuse": "false",
     "spark.driver.host": "127.0.0.1",
     "spark.task.maxFailures": "1",
-    "spark.driver.memory": "5g",
+    "spark.driver.memory": "128g",
```
Comment (eordentlich) [inline python/tests/conftest.py:66]: Will this severely limit where this can run?

File: python/tests/test_logistic_regression.py [hunk 2]
```diff
@@ -2092,3 +2099,58 @@ def test_quick_double_precision(
         reg_factors=reg_factors,
         float32_inputs=float32_inputs,
     )
+
+
+@pytest.mark.slow
+def test_sparse_int64() -> None:
+
+    gpu_number = 1
+    data_shape = (int(1e7), 30)
```
Comment (eordentlich) [inline python/tests/test_logistic_regression.py:2108]: Will this give only 300 million nonzero?   Thats below the max int limit to be tested?

**Comment by lijinf2** [comment]
build

File: python/tests/conftest.py (hunk 1 above)
Comment (lijinf2) [reply python/tests/conftest.py]: Revised to 32g. The added test leveraged disk and finished in about 10 mins. Seems acceptable as the dataset itself is large. 

Tried setting this to 960g, while physical memory has only 128g. All tests in test_logistic_regression.py run fine. Seems a larger value did not limit the tests to run.

File: python/tests/test_logistic_regression.py (hunk 2 above)
Comment (lijinf2) [reply python/tests/test_logistic_regression.py]: Revised to use larger dataset and SparseDataGen.

**Comment by lijinf2** [comment]
build

File: python/tests/test_logistic_regression.py [hunk 3]
```diff
@@ -2092,3 +2108,110 @@ def test_quick_double_precision(
...
+    data_gen = SparseRegressionDataGen(data_gen_args)
+    df, _, _ = data_gen.gen_dataframe_and_meta(conftest._spark)
+    df = df.cache()
+
+    # check nnz > int32.max
+    def get_nnz(sparse_vec: SparseVector) -> int:
+        return len(sparse_vec.values)
+
+    udf_get_nnz = udf(get_nnz, LongType())
+    nnz_df = df.select(udf_get_nnz("features").alias("nnz"))
```
Comment (eordentlich) [inline python/tests/test_logistic_regression.py:2190]: Is this slow?  udf is invoked on each row I believe.

File: python/tests/test_logistic_regression.py [hunk 4]
```diff
@@ -1,10 +1,20 @@
+import math
+import os
+import sys
+
+file_path = os.path.abspath(__file__)
```
Comment (eordentlich) [inline python/tests/test_logistic_regression.py:5]: Would it be better/cleaner to set PYTHONPATH env var in run_test.sh .

File: python/tests/test_logistic_regression.py (hunk 3 above)
Comment (lijinf2) [reply python/tests/test_logistic_regression.py]: Good catch. I replaced it with mapInPandas. Runtime was reduced by 45 seconds.

File: python/tests/test_logistic_regression.py (hunk 4 above)
Comment (lijinf2) [reply python/tests/test_logistic_regression.py]: Yeah, revised to use PYTHONPATH env variable.

**Comment by lijinf2** [comment]
build

**Review by eordentlich — APPROVED** [review]
👍

**Pull request merged**